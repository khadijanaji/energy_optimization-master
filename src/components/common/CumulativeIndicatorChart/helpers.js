/* eslint-disable no-magic-numbers, react/no-multi-comp, max-len, max-lines */
// @flow


import moment from "moment";
import __, {isNumber, range, isNull} from "lodash";
import * as React from "react";

import strings from "./strings";
import loadingIcon from "../../../img/loading.svg";

export const recordBeginning = 7;
export const dayHours = 24;
export const precision = 3;

export const TIMELINETYPE = {
  "DAILY": "DAILY",
  "MONTHLY": "MONTHLY",
  "YEARLY": "YEARLY",
  "CUSTOM_MONTHLY": "CUSTOM_MONTHLY",
  "CUSTOM_YEARLY": "CUSTOM_YEARLY"
};

export const getYesterday = (): moment => moment().subtract(1, "days");

export const getMonthsOfThisYear = (date: moment): Array<string> => {
  const thisMonth = date.month() + 1;
  const tmp = date.clone();
  return range(0, thisMonth).map((ind) => (tmp.month(ind).format("MMM")));
};

export const getMonthsOfRange = (fromDate: moment, toDate: moment): Array<string> => {
  const fromMonth = fromDate.month() + 1;
  const toMonth = toDate.month() + 1;
  const tmp = fromDate.clone();
  return range(fromMonth, toMonth).map((index) => (tmp.month(index).format("MMM")));
};

export const getMonthsOfRanges = (fromDate: moment, toDate: moment): Array<string> => {
  const output = [];
  const dateCursor = fromDate.clone();
  while (dateCursor.isSameOrBefore(toDate)) {
    output.push(dateCursor.format("MMM"));
    dateCursor.add(1, "days");
  }
  return output;
};

export const getFirstDayOfThisMonth = (yesterday: moment): moment => (yesterday.
  clone().
  startOf("month"));

export const getDaysOfMonth = (dayOfMonth: moment): Array<string> => {
  const firstDay = getFirstDayOfThisMonth(dayOfMonth);
  const dayNumber = dayOfMonth.date();
  return range(0, dayNumber).map((ind) => (firstDay.
    clone().
    add(ind, "days").
    format("DD")));
};

export const getDaysOfMonthBetween = (fromDate: moment, toDate: moment): Array<string> => {
  const output = [];
  const dateCursor = fromDate.clone();
  while (dateCursor.isSameOrBefore(toDate)) {
    output.push(dateCursor.format("DD"));
    dateCursor.add(1, "days");
  }
  return output;
};

export const getDatesBetween = (fromDate: moment, toDate: moment): Array<string> => {
  const output = [];
  const dateCursor = fromDate.clone();
  while (dateCursor.isSameOrBefore(toDate)) {
    output.push(dateCursor.format("DD/MM"));
    dateCursor.add(1, "days");
  }
  return output;
};

export const getTargetsFromRecordBeginningToCurrentMonth = (yearlyTarget: number, currentMonth: number) => {
  const monthsInYear = 12;
  const monthlyTarget = yearlyTarget / monthsInYear;

  return __.range(1, currentMonth + 2).map((elem) => (Math.round(elem * monthlyTarget)));
};

export const getTargetsFromRecordBeginningToCurrentDay = (monthlyTarget: number, currentDayOfMonth: moment) => {
  const currentDay = currentDayOfMonth.date();
  const daysInmonth = currentDayOfMonth.daysInMonth();
  const dailyTarget = monthlyTarget / daysInmonth;

  return __.range(1, currentDay + 1).map((elem) => (Math.round(elem * dailyTarget)));
};

export const getTodayProductionHours = () => {
  const now = moment().hour();
  return (now >= recordBeginning)
    ? (now - recordBeginning + 1)
    : (dayHours - recordBeginning + now);
};

export const getHoursFromRecordBeginningToCurrentHour = () => {
  const now = moment().hour();
  if (now >= recordBeginning) {
    return __.
      range(recordBeginning, now + 1).
      map((elem) => `${elem} h`);
  }
  return __.
    range(recordBeginning, dayHours).
    concat(__.range(0, now + 1)).
    map((elem) => `${elem} h`);
};

export const getTargetsFromRecordBeginningToCurrentHour = (target: number) => {
  const productionHours = getTodayProductionHours();
  const hourlyTarget = target / dayHours;
  const targets = __.range(0, productionHours).map((elem) => (Math.round(elem * hourlyTarget)));
  targets[0] = 0;
  return targets;
};

export const getActualTargetMonth = (monthlyTarget: number) => {
  const currentDay = getYesterday().date();
  const daysInmonth = getYesterday().daysInMonth();
  if (monthlyTarget) {
    return (monthlyTarget / daysInmonth) * currentDay;
  }
  return (<span className="cumulative-indicator-chart__container__value__UNAVAILABLE">{"Donnée non disponible"}</span>);
};

export const getActualTarget = (target: number) => {
  const productionHours = getTodayProductionHours();
  if (target) {
    const ratio = (productionHours - 1) / dayHours;
    return target * ratio;
  }
  return (<span className="cumulative-indicator-chart__container__value__UNAVAILABLE">{"Donnée non disponible"}</span>);
};

export const getGapBetweenActualAndTargetValue = (target: number, value: number) => {
  if (target) {
    return value - target;
  }
  return (<span className="cumulative-indicator-chart__container__value__UNAVAILABLE">{"Donnée non disponible"}</span>);
};

export const getGapPercent =
    (target: number, value: number) => {
      if (target) {
        return (100.0 * parseFloat(getGapBetweenActualAndTargetValue(target, value)) / target);
      }
      return (<span className="cumulative-indicator-chart__container__value__UNAVAILABLE">
        {"Donnée non disponible"}
      </span>);
    };

type mapChartDataType = {
  title: string,
  subtitle: string,
  unit: string,
  target?: Array<number>,
  values: Array<number>,
  timeLineType: string,
  fromDate?: moment,
  toDate?: moment
};

export const formatValues = (values: Array<number>) => {
  if (values) {
    return values.map((value) => value && parseFloat(Math.round(value * 1000) / 1000));
  }
  return values;
};

export const formatYearlyValues = (values: Array<number>, names: Array<string>) => {
  if (values) {
    return values.map((value, index) => ({
      "name": names[index],
      "y": value && parseFloat(Math.round(value * 1000) / 1000)
    }));
  }
  return values;
};

export const mapValuesToCumulativeValues =
    (values: Array<number>): Array<number> => {
      let sum = 0;
      return values && values.map((value) => (sum += value));
    };

export const mapDailyChartData = ({title, subtitle, unit, target, values}: mapChartDataType) => {
  const categories = getHoursFromRecordBeginningToCurrentHour();
  const data = target && getTargetsFromRecordBeginningToCurrentHour(target[0]);
  const allValues = values && [].concat(0, ...values);
  return {
    "chart": {
      "type": "spline",
      "zoomType": "xy",
      "pinchType": "xy"
    },
    "title": {"text": title},
    "subtitle": {"text": subtitle},
    "xAxis": {categories},
    "yAxis": {"title": {"text": unit}},
    "plotOptions": {
      "line": {
        "dataLabels": {"enabled": true},
        "enableMouseTracking": false
      }
    },
    "tooltip": {
      "formatter" () {
        return `<span>${this.x.toString()}</span> : <b style="color:${this.color}">${this.y.toString()}</b>`;
      },
      "useHTML": true
    },
    "series": [
      {
        "name": strings.target,
        data,
        "dashStyle": "Dot",
        "color": "#A8FBA8"
      },
      {
        "name": strings.actuals,
        "data": formatValues(allValues)
      }
    ]
  };
};

export const mapMonthlyChartData = ({title, subtitle, unit, target, values}: mapChartDataType) => {
  const categories = getDaysOfMonth(getYesterday());
  const data = target && getTargetsFromRecordBeginningToCurrentDay(target[0], getYesterday());
  return {
    "chart": {
      "type": "spline",
      "zoomType": "xy",
      "pinchType": "xy"
    },
    "title": {"text": title},
    "subtitle": {"text": subtitle},
    "xAxis": {categories},
    "yAxis": {"title": {"text": unit}},
    "plotOptions": {
      "line": {
        "dataLabels": {"enabled": true},
        "enableMouseTracking": false
      }
    },
    "tooltip": {
      "formatter" () {
        return `<span>${this.x.toString()}</span> : <b style="color:${this.color}">${this.y.toString()}</b>`;
      },
      "useHTML": true
    },
    "series": [
      {
        "name": strings.target,
        data,
        "dashStyle": "Dot",
        "color": "#A8FBA8"
      },
      {
        "name": strings.actuals,
        "data": formatValues(mapValuesToCumulativeValues(values))
      }
    ]
  };
};

const replaceValuesWithNull = (values: Array<number>) => values.map((value) => {
  if (isNull(value)) {
    return 0;
  }
  return null;
});

const formatTooltipValue = (color: string, value: string) => {
  if (color === "red") {
    return "Valeur non disponible";
  }
  return value;
};

const rangeHistoricTooltip = (dates: Array<string>) => ({
  "formatter" () {
    return `<span>${dates[this.series.data.indexOf(this.point)]}</span> : <b style="color:${this.color}">${formatTooltipValue(this.color, this.y.toString())}</b>`;
  },
  "useHTML": true
});

export const mapMonthlyRangeChartData = ({title, subtitle, unit, values, fromDate, toDate, target}: mapChartDataType) => {
  const categories = getDaysOfMonthBetween(fromDate, toDate);
  const dates = getDatesBetween(fromDate, toDate);
  return {
    "chart": {
      "type": "spline",
      "zoomType": "xy",
      "pinchType": "xy"
    },
    "title": {"text": title},
    "subtitle": {"text": subtitle},
    "xAxis": {
      categories,
      "gridLineWidth": "1",
      "lineWidth": 1,
      "tickmarkPlacement": "on"
    },
    "yAxis": {"title": {"text": unit}},
    "plotOptions": {
      "line": {
        "dataLabels": {"enabled": true},
        "enableMouseTracking": false
      }
    },
    "tooltip": rangeHistoricTooltip(dates),
    "series": [
      {
        "name": "Valeurs non disponible",
        "data": replaceValuesWithNull(values),
        "color": "red"
      },
      {
        "name": strings.actuals,
        "data": formatValues(values)
      },
      {
        "name": strings.target,
        "data": target,
        "dashStyle": "ShortDash",
        "color": "#A8FBA8"
      }
    ]
  };
};

const positionOfUniqueMonths = (months: string[]) => {
  const result = [0];
  for (let index = 1; index < months.length; index += 1) {
    if (months[index] !== months[index - 1]) {
      result.push(index);
    }
  }
  return result;
};

export const mapYearlyRangeChartData = ({title, subtitle, unit, values, fromDate, toDate, target}: mapChartDataType) => {
  const categories = getMonthsOfRanges(fromDate, toDate);
  const dates = getDatesBetween(fromDate, toDate);
  return {
    "chart": {
      "type": "spline",
      "zoomType": "xy",
      "pinchType": "xy"
    },
    "title": {"text": title},
    "subtitle": {"text": subtitle},
    "xAxis": {
      categories,
      "tickPositions": positionOfUniqueMonths(categories),
      "gridLineWidth": "1",
      "lineWidth": 1,
      "tickmarkPlacement": "on",
      "isCustomYearly": true
    },
    "yAxis": {"title": {"text": unit}},
    "plotOptions": {
      "line": {
        "dataLabels": {"enabled": true},
        "enableMouseTracking": false
      }
    },
    "tooltip": rangeHistoricTooltip(dates),
    "series": [
      {
        "name": "Valeurs non disponible",
        "data": replaceValuesWithNull(values),
        "color": "red",
        "lineWidth": 0
      },
      {
        "name": strings.actuals,
        "data": formatYearlyValues(values, categories)
      },
      {
        "name": strings.target,
        "data": target,
        "dashStyle": "ShortDash",
        "color": "#A8FBA8"
      }
    ]
  };
};

export const mapYearlyChartData = ({title, subtitle, unit, target, values}: mapChartDataType) => {
  const categories = getMonthsOfThisYear(getYesterday());
  return {
    "chart": {
      "type": "column",
      "zoomType": "xy",
      "pinchType": "xy"
    },
    "title": {"text": title},
    "subtitle": {"text": subtitle},
    "xAxis": {categories},
    "yAxis": {"title": {"text": unit}},
    "plotOptions": {
      "line": {
        "dataLabels": {"enabled": true},
        "enableMouseTracking": false
      }
    },
    "tooltip": {
      "formatter" () {
        return `<span>${this.x.toString()}</span> : <b style="color:${this.color}">${this.y.toString()}</b>`;
      },
      "useHTML": true
    },
    "series": [
      {
        "name": strings.target,
        "data": target,
        "dashStyle": "Dot",
        "color": "#A8FBA8"
      },
      {
        "name": strings.actuals,
        "data": formatValues(values)
      }
    ]
  };
};

export const mapChartData = (mapchartDataParams: mapChartDataType) => {
  switch (mapchartDataParams.timeLineType) {
  case TIMELINETYPE.MONTHLY:
    return mapMonthlyChartData(mapchartDataParams);
  case TIMELINETYPE.YEARLY:
    return mapYearlyChartData(mapchartDataParams);
  case TIMELINETYPE.CUSTOM_MONTHLY:
    return mapMonthlyRangeChartData(mapchartDataParams);
  case TIMELINETYPE.CUSTOM_YEARLY:
    return mapYearlyRangeChartData(mapchartDataParams);
  case TIMELINETYPE.DAILY:
  default:
    return mapDailyChartData(mapchartDataParams);
  }
};

export const renderValue = (value: any, status: string) => {
  if (status === "Loading") {
    return (
      <img
        alt={"chargement"}
        className="category-indicator-item__loading"
        src={loadingIcon}
      />
    );
  }
  if (isNumber(value)) {
    return (parseFloat(value).toFixed(precision));
  }
  return (<span className="cumulative-indicator-chart__container__value__UNAVAILABLE">{"Donnée non disponible"}</span>);
};

export const getGravityClassName = (actualTarget: number, value: number) => (value && ((value >= actualTarget)
  ? "cumulative-indicator-chart__container__value__OK"
  : "cumulative-indicator-chart__container__value__DANGER"));
