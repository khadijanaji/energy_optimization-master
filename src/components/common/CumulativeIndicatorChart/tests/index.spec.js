/* eslint-disable no-magic-numbers */
/* globals describe it expect */

import {shallow} from "enzyme";
import React from "react";
import moment from "moment";

import {TIMELINETYPE} from "../helpers";

jest.mock("lodash", () => ({
  "random": () => 12345678,
  "range": () => [
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19
  ],
  "isNumber": () => true
}));

const props = {
  "title": "title",
  "subtitle": "subtitle",
  "unit": "unit",
  "target": [223.45],
  "values": [
    10.2,
    25,
    48,
    68.84,
    98,
    99.48,
    120,
    140.2,
    180.1,
    194.02,
    200.34,
    200.14,
    208.45
  ],
  "status": "Success",
  "timeLineType": TIMELINETYPE.DAILY

};

jest.resetModules();
jest.doMock("moment", () => () => (moment("20190419 15")));
const CumulativeIndicatorChart = require("../");

describe("Component", () => {
  describe("CumulativeIndicatorChart Render", () => {
    it("With daily time line, should match snapshot", () => {
      const wrapper = shallow(<CumulativeIndicatorChart.default
        {...props}
      />);
      expect(wrapper).toMatchSnapshot();
    });
    it("With daily time line and null values, should match snapshot", () => {
      const wrapper = shallow(<CumulativeIndicatorChart.default
        {...{
          ...props,
          "values": null
        }}
      />);
      expect(wrapper).toMatchSnapshot();
    });
    it("With monthly time line, should match snapshot", () => {
      const wrapper = shallow(<CumulativeIndicatorChart.default
        {...{
          ...props,
          "timeLineType": TIMELINETYPE.MONTHLY
        }}
      />);
      expect(wrapper).toMatchSnapshot();
    });
    it("With yearly time line, should match snapshot", () => {
      const wrapper = shallow(<CumulativeIndicatorChart.default
        {...{
          ...props,
          "timeLineType": TIMELINETYPE.YEARLY
        }}
      />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
