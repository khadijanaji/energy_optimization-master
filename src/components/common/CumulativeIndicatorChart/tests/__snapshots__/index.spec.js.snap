// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CumulativeIndicatorChart Render With daily time line and null values, should match snapshot 1`] = `
<div>
  <Highcharts
    container="HighChart-12345678"
    options={
      Object {
        "chart": Object {
          "pinchType": "xy",
          "type": "spline",
          "zoomType": "xy",
        },
        "plotOptions": Object {
          "line": Object {
            "dataLabels": Object {
              "enabled": true,
            },
            "enableMouseTracking": false,
          },
        },
        "series": Array [
          Object {
            "color": "#A8FBA8",
            "dashStyle": "Dot",
            "data": Array [
              0,
              74,
              84,
              93,
              102,
              112,
              121,
              130,
              140,
              149,
              158,
              168,
              177,
            ],
            "name": "Objectif",
          },
          Object {
            "data": null,
            "name": "Valeurs actuelles",
          },
        ],
        "subtitle": Object {
          "text": "subtitle",
        },
        "title": Object {
          "text": "title",
        },
        "tooltip": Object {
          "formatter": [Function],
          "useHTML": true,
        },
        "xAxis": Object {
          "categories": Array [
            "7 h",
            "8 h",
            "9 h",
            "10 h",
            "11 h",
            "12 h",
            "13 h",
            "14 h",
            "15 h",
            "16 h",
            "17 h",
            "18 h",
            "19 h",
          ],
        },
        "yAxis": Object {
          "title": Object {
            "text": "unit",
          },
        },
      }
    }
    status="Success"
  />
  <div
    className="cumulative-indicator-chart__root-container"
  >
    <table
      border="1"
      className="cumulative-indicator-chart__container"
    >
      <tbody>
        <tr>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__border-right"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Valeur actuelle:
            </div>
            <div
              className={null}
            >
              NaN
               
            </div>
          </td>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__target"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Cible à l'instant:
            </div>
            <div
              className="cumulative-indicator-chart__container__value"
            >
              74.483
               
              unit
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__border-top"
            colSpan="2"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Écart:
            </div>
            <div
              className="cumulative-indicator-chart__container__value"
            >
              <span>
                -74.483
                 
                unit
              </span>
              <span>
                <span
                  className={null}
                >
                   (
                  -100.000
                   %)
                </span>
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Component CumulativeIndicatorChart Render With daily time line, should match snapshot 1`] = `
<div>
  <Highcharts
    container="HighChart-12345678"
    options={
      Object {
        "chart": Object {
          "pinchType": "xy",
          "type": "spline",
          "zoomType": "xy",
        },
        "plotOptions": Object {
          "line": Object {
            "dataLabels": Object {
              "enabled": true,
            },
            "enableMouseTracking": false,
          },
        },
        "series": Array [
          Object {
            "color": "#A8FBA8",
            "dashStyle": "Dot",
            "data": Array [
              0,
              74,
              84,
              93,
              102,
              112,
              121,
              130,
              140,
              149,
              158,
              168,
              177,
            ],
            "name": "Objectif",
          },
          Object {
            "data": Array [
              0,
              10.2,
              25,
              48,
              68.84,
              98,
              99.48,
              120,
              140.2,
              180.1,
              194.02,
              200.34,
              200.14,
              208.45,
            ],
            "name": "Valeurs actuelles",
          },
        ],
        "subtitle": Object {
          "text": "subtitle",
        },
        "title": Object {
          "text": "title",
        },
        "tooltip": Object {
          "formatter": [Function],
          "useHTML": true,
        },
        "xAxis": Object {
          "categories": Array [
            "7 h",
            "8 h",
            "9 h",
            "10 h",
            "11 h",
            "12 h",
            "13 h",
            "14 h",
            "15 h",
            "16 h",
            "17 h",
            "18 h",
            "19 h",
          ],
        },
        "yAxis": Object {
          "title": Object {
            "text": "unit",
          },
        },
      }
    }
    status="Success"
  />
  <div
    className="cumulative-indicator-chart__root-container"
  >
    <table
      border="1"
      className="cumulative-indicator-chart__container"
    >
      <tbody>
        <tr>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__border-right"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Valeur actuelle:
            </div>
            <div
              className="cumulative-indicator-chart__container__value__OK"
            >
              208.450
               
              unit
            </div>
          </td>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__target"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Cible à l'instant:
            </div>
            <div
              className="cumulative-indicator-chart__container__value"
            >
              74.483
               
              unit
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__border-top"
            colSpan="2"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Écart:
            </div>
            <div
              className="cumulative-indicator-chart__container__value"
            >
              <span>
                133.967
                 
                unit
              </span>
              <span>
                <span
                  className="cumulative-indicator-chart__container__value__OK"
                >
                   (
                  179.861
                   %)
                </span>
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Component CumulativeIndicatorChart Render With monthly time line, should match snapshot 1`] = `
<div>
  <Highcharts
    container="HighChart-12345678"
    options={
      Object {
        "chart": Object {
          "pinchType": "xy",
          "type": "spline",
          "zoomType": "xy",
        },
        "plotOptions": Object {
          "line": Object {
            "dataLabels": Object {
              "enabled": true,
            },
            "enableMouseTracking": false,
          },
        },
        "series": Array [
          Object {
            "color": "#A8FBA8",
            "dashStyle": "Dot",
            "data": Array [
              52,
              60,
              67,
              74,
              82,
              89,
              97,
              104,
              112,
              119,
              127,
              134,
              142,
            ],
            "name": "Objectif",
          },
          Object {
            "data": Array [
              10.2,
              35.2,
              83.2,
              152.04,
              250.04,
              349.52,
              469.52,
              609.72,
              789.82,
              983.84,
              1184.18,
              1384.32,
              1592.77,
            ],
            "name": "Valeurs actuelles",
          },
        ],
        "subtitle": Object {
          "text": "subtitle",
        },
        "title": Object {
          "text": "title",
        },
        "tooltip": Object {
          "formatter": [Function],
          "useHTML": true,
        },
        "xAxis": Object {
          "categories": Array [
            "08",
            "09",
            "10",
            "11",
            "12",
            "13",
            "14",
            "15",
            "16",
            "17",
            "18",
            "19",
            "20",
          ],
        },
        "yAxis": Object {
          "title": Object {
            "text": "unit",
          },
        },
      }
    }
    status="Success"
  />
  <div
    className="cumulative-indicator-chart__root-container"
  >
    <table
      border="1"
      className="cumulative-indicator-chart__container"
    >
      <tbody>
        <tr>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__border-right"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Valeur actuelle:
            </div>
            <div
              className="cumulative-indicator-chart__container__value__OK"
            >
              1592.770
               
              unit
            </div>
          </td>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__target"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Cible à l'instant:
            </div>
            <div
              className="cumulative-indicator-chart__container__value"
            >
              134.070
               
              unit
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__border-top"
            colSpan="2"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Écart:
            </div>
            <div
              className="cumulative-indicator-chart__container__value"
            >
              <span>
                1458.700
                 
                unit
              </span>
              <span>
                <span
                  className="cumulative-indicator-chart__container__value__OK"
                >
                   (
                  1088.014
                   %)
                </span>
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Component CumulativeIndicatorChart Render With yearly time line, should match snapshot 1`] = `
<div>
  <Highcharts
    container="HighChart-12345678"
    options={
      Object {
        "chart": Object {
          "pinchType": "xy",
          "type": "column",
          "zoomType": "xy",
        },
        "plotOptions": Object {
          "line": Object {
            "dataLabels": Object {
              "enabled": true,
            },
            "enableMouseTracking": false,
          },
        },
        "series": Array [
          Object {
            "color": "#A8FBA8",
            "dashStyle": "Dot",
            "data": Array [
              223.45,
            ],
            "name": "Objectif",
          },
          Object {
            "data": Array [
              10.2,
              25,
              48,
              68.84,
              98,
              99.48,
              120,
              140.2,
              180.1,
              194.02,
              200.34,
              200.14,
              208.45,
            ],
            "name": "Valeurs actuelles",
          },
        ],
        "subtitle": Object {
          "text": "subtitle",
        },
        "title": Object {
          "text": "title",
        },
        "tooltip": Object {
          "formatter": [Function],
          "useHTML": true,
        },
        "xAxis": Object {
          "categories": Array [
            "Aug",
            "Sep",
            "Oct",
            "Nov",
            "Dec",
            "Jan",
            "Feb",
            "Mar",
            "Apr",
            "May",
            "Jun",
            "Jul",
            "Aug",
          ],
        },
        "yAxis": Object {
          "title": Object {
            "text": "unit",
          },
        },
      }
    }
    status="Success"
  />
  <div
    className="cumulative-indicator-chart__root-container"
  >
    <table
      border="1"
      className="cumulative-indicator-chart__container"
    >
      <tbody>
        <tr>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__border-right"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Valeur actuelle:
            </div>
            <div
              className="cumulative-indicator-chart__container__value__OK"
            >
              1592.770
               
              unit
            </div>
          </td>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__target"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Cible à l'instant:
            </div>
            <div
              className="cumulative-indicator-chart__container__value"
            >
              223.450
               
              unit
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="cumulative-indicator-chart__container__value cumulative-indicator-chart__container__border-top"
            colSpan="2"
          >
            <div
              className="cumulative-indicator-chart__label"
            >
              Écart:
            </div>
            <div
              className="cumulative-indicator-chart__container__value"
            >
              <span>
                1369.320
                 
                unit
              </span>
              <span>
                <span
                  className="cumulative-indicator-chart__container__value__OK"
                >
                   (
                  612.808
                   %)
                </span>
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
